src/ShopBundle/objects/TShopBasket/TShopBasketArticleCoreList.class.php

```diff
@@ -801,4 +801,31 @@ protected function PostDeleteItemHook($oDeletedItem)
             $this->aObservers[$sObserverName]->OnBasketItemDeleteEvent($oDeletedItem);
         }
     }
+
+    /**
+     * items with the same sBasketItemKey will be merged into one item.
+     */
+    public function mergeIdenticalBasketItems(): void
```

Comment:
Does this TShopBasketArticleCoreList reflect database entities?

If so wouldn't you need to update the db as well?

Reply:
nope - the basket is kept in memory/session.